### DISCUSSION TITLE
Image | NanoPi R5C

### DISCUSSION THREAD

**Discussion by buterrip69** [MichaIng/DietPi#5999] [Image Request]
### Creating an image request
 #### Formal device information
 
     * Device name | NanoPi R5C
 
     * Official product URL | https://www.friendlyelec.com/index.php?route=product/product&product_id=290
 
     * GitHub resource URL (if available) | https://github.com/friendlyarm/friendlywrt
 
     * Image download URL | https://download.friendlyelec.com/NanoPiR5C
 
 
 #### Is the SBC officially supported by the Debian installer?
 
     * not sure
 
 
 #### If not, is a reliable 3rd party Debian image available for this SBC?
 
     * ...
 
 
 #### If not, are there install instructions for Debian available?
 
     * no
 
 
I've got a R5c, and i would like to use dietpi.
Since R5c and R5s are very similar, can i flash r5s image on r5c?
Otherwise, can it be made?

**Comment by Joulinar** [comment]
hmm you could at least give it a try.

**Reply by buterrip69** [reply]
After checking that even friendlyelec is using the same images for both, i did try.
As expected, it does work.

**Reply by MichaIng** [reply]
Indeed, the FriendlyELEC images (we use same bootloader and kernel) are for "NanoPi 5 series", so should work on all of their RK3568 SBCs, which is quite handy.

Are the Ethernet LEDs correctly linked to the interfaces. E.g. if you attach and detach Ethernet cables to each Ethernet port, are the LAN and WAN LEDs correctly indicating cable connection state and in case RX/TX (blinking)?

**Reply by buterrip69** [reply]
Hello, matter of fact the ports are not working, lan is not recognized, and wan is apparently trying to lease or get dhcp, so wan must be working.
Leds by the doors are not working either.
I believe this is due to r5s having two lan's and one wan and r5c one lan and one wan.
Maybe try their's updated kernel?
Cause they made a new one when r5c came out.
I do have grafics though.

**Reply by MichaIng** [reply]
Makes  sense. DietPi by default configures DHCP on `eth0`, which is the WAN port on NanoPi R5S at least. The LAN port interface `eth1` would currently need to be configured manually, e.g. with a dedicated ifupdown config in `/etc/network/interfaces.d`.

**Reply by tgp-2** [reply]
another R5C user here ... using the R5S image, similar observations as noted above, willing to try and help with basic device testing

**Reply by MichaIng** [reply]
So to verify:
- When having (Internet facing) Ethernet attached to the WAN port, it gets configured with DHCP successfully?
- If so, is also the WAN port LED (the one on the PCB/case, not the ones one port itself) lid and blinking on network I/O?

**Reply by tgp-2** [reply]
correct (yes for both items above).

also, with WiFi connection active (and WAN port unplugged) ... WAN LED and WL LED are both off.

**Reply by MichaIng** [reply]
So then it seems we can use the very same image for both SBCs. Shall I rename the hardware model to "NanoPi R5S/R5C" or shall we host two (for now identical) images in case we need to differenciate in the future?

**Reply by buterrip69** [reply]
Hello, after further testing i can confirm all of the above.

Since friendlyelec also use the same images for both, i believe you don't need to differenciate either. 

Only hardware difference i see is 3 ethernets r5s doors and 2 ethernet doors on the r5c.

Also if you want to flash dietpi into mmc you just use dd to copy it from a usb and it works even faster as expected.

Anyway i'm waiting on another little guy to use with dietpi with kodi or something alike, this one will stay at openwrt for the use of wireguard at full network speed.

If in any case you need further testing just hit me up i can just insert the tf card ;)

**Reply by tgp-2** [reply]
it does look like a single shared image might be ok (I'm not familiar with what differences would normally require a separate image)

generally the R5C feature set is a subset of the R5S ... and I see only a few other differences (LEDs mapped to different functions, different M.2 key/support, USB 3.2 gen1 vs USB 3.0 ports)

**Reply by MichaIng** [reply]
Okay, I renamed the hardware model internally to indicate that both models are supported. May rename it to "NanoPi 5 series" if further variants are released. Same will be done on website and docs:
- https://github.com/MichaIng/DietPi-Website/issues/248
- https://github.com/MichaIng/DietPi-Docs/issues/819

**Reply by tgp-2** [reply]
What will "device model" show in welcome banner?

**Reply by MichaIng** [reply]
"NanoPi R5S/R5C"
